src/unix.rs

```diff
@@ -130,6 +132,18 @@ impl MmapInner {
         }
     }
 
+    pub fn set_protection(&mut self, prot: Protection) -> io::Result<()> {
+        let result = unsafe { libc::mprotect(self.ptr,
```

Comment:
Or maybe it could do the same thing as the jit test and go from read+write -> read+execute, then execute it?